### PR TITLE
Introduce physical plan and add switch

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: ref #4739

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by SeaRise** [comment]
wait for https://github.com/pingcap/tiflash/pull/5034

File: .github/workflows/license-checker.yml [hunk 1]
```diff
@@ -3,10 +3,10 @@ name: License checker
 on:
   push:
     branches:
-      - master
+      - planner_refactory
   pull_request:
     branches:
-      - master
+      - planner_refactory
```
Comment (SeaRise) [inline .github/workflows/license-checker.yml:9]: modify for license checker.
I'll revert to `master` at the last pr of branch `planner refactory`

File: dbms/src/Flash/Planner/Planner.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    void executeImpl(DAGPipeline & pipeline);
+
+private:
+    Context & context;
+
+    std::vector<BlockInputStreams> input_streams_vec;
+
+    const DAGQueryBlock & query_block;
+
+    /// How many streams we ask for storage to produce, and in how many threads we will do further processing.
```
Comment (ywqzzy) [inline dbms/src/Flash/Planner/Planner.h:55]: Is it related to mpp receiver？

File: dbms/src/Flash/Planner/optimize.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        return plan;
+    }
+
+    static RulePtr create() { return std::make_shared<FinalizeRule>(); }
+};
+
+PhysicalPlanPtr optimize(const Context & context, PhysicalPlanPtr plan)
+{
+    assert(plan);
+    static std::vector<RulePtr> rules{FinalizeRule::create()};
```
Comment (ywqzzy) [inline dbms/src/Flash/Planner/optimize.cpp:44]:
```suggestion
    static std::vector<RulePtr> rules(FinalizeRule::create());
```
nit: May be same code style with other files.

File: dbms/src/Flash/tests/gtest_planner_interpreter.cpp [hunk 4]
```diff
@@ -19,13 +19,15 @@ namespace DB
 {
 namespace tests
 {
-class InterpreterExecuteTest : public DB::tests::ExecutorTest
+class PlannerInterpreterExecuteTest : public DB::tests::ExecutorTest
```
Comment (SeaRise) [inline dbms/src/Flash/tests/gtest_planner_interpreter.cpp:22]: input of `gtest_planner_interpreter` is same as `gtest_qb_interpreter`

File: dbms/src/Flash/Planner/optimize.cpp (hunk 3 above)
Comment (SeaRise) [reply dbms/src/Flash/Planner/optimize.cpp:44]: it will build fail.

File: dbms/src/Flash/Planner/Planner.h (hunk 2 above)
Comment (SeaRise) [reply dbms/src/Flash/Planner/Planner.h]: changed to `How many streams we will do processing.`

File: dbms/src/Flash/Planner/PlanType.h [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+struct PlanType
+{
+    enum PlanTypeEnum
+    {
+        Source = 0x1,
+    };
+    PlanTypeEnum enum_value;
+
+    PlanType(int value = 0) // NOLINT(google-explicit-constructor)
+        : enum_value(static_cast<PlanTypeEnum>(value))
```
Comment (windtalker) [inline dbms/src/Flash/Planner/PlanType.h:30]: It seems the default value of `value` is not a valid PlanTypeEnum?

File: dbms/src/Flash/Planner/Planner.h [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    void executeImpl(DAGPipeline & pipeline);
+
+private:
+    Context & context;
+
+    std::vector<BlockInputStreams> input_streams_vec;
+
+    const DAGQueryBlock & query_block;
+
+    /// How many streams we will do processing.
```
Comment (windtalker) [inline dbms/src/Flash/Planner/Planner.h:55]: The comments should be 
"Max streams we will do processing"?

File: dbms/src/Flash/Coprocessor/InterpreterDAG.cpp [hunk 7]
```diff
@@ -65,12 +66,24 @@ BlockInputStreams InterpreterDAG::executeQueryBlock(DAGQueryBlock & query_block)
...
     }
-    DAGQueryBlockInterpreter query_block_interpreter(
-        context,
-        input_streams_vec,
-        query_block,
-        max_streams);
-    return query_block_interpreter.execute();
+    if (context.getSettingsRef().enable_planner && Planner::isSupported(query_block))
+    {
+        Planner planner(
```
Comment (windtalker) [inline dbms/src/Flash/Coprocessor/InterpreterDAG.cpp:71]: planner still need `query_block` as its input?

Comment (SeaRise) [reply dbms/src/Flash/Coprocessor/InterpreterDAG.cpp:71]: After all operators are converted to physical plan, we can get out of the query block.

Comment (SeaRise) [reply dbms/src/Flash/Coprocessor/InterpreterDAG.cpp:71]: In this way, the operators can be converted into physical plan one by one

**Review by windtalker — APPROVED** [review]
LGTM

**Review by ywqzzy — APPROVED** [review]

**Pull request merged**